### PR TITLE
Initial commit of istio refactor

### PR DESCRIPTION
**Pull request by jrsdav**
Related: https://github.com/streamnative/cloud-api-server/issues/800
This refactor removes the locally sourced Istio Operator helm chart in favor of of the recently re-worked Istio configuration [stored in the `streamnative/terraform-helm-charts` repo](https://github.com/streamnative/terraform-helm-charts).

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -33,11 +33,9 @@ locals {
...
 
   ### Istio Config
-  default_sources          = ["service", "ingress"]
-  istio_namespace          = var.istio_namespace == "istio-system" ? "istio-system" : var.istio_namespace
-  istio_operator_namespace = var.istio_operator_namespace == "istio-operator" ? "istio-operator" : var.istio_operator_namespace
-  istio_sources            = ["istio-gateway", "istio-virtualservice"]
-  sources                  = var.enable_istio_operator ? concat(local.istio_sources, local.default_sources) : local.default_sources
+  default_sources = ["service", "ingress"]
+  istio_sources   = ["istio-gateway", "istio-virtualservice"]
+  sources         = var.enable_istio ? concat(local.istio_sources, local.default_sources) : local.default_sources
```
Comment (EronWright) [inline locals.tf:38]: How about we cease with the anti-pattern of defining all locals into one file.  I feel that these locals belong in external_dns.tf.

File: external_dns.tf [hunk 2]
```diff
@@ -96,7 +96,7 @@ resource "helm_release" "external_dns" {
   repository      = var.external_dns_helm_chart_repository
   timeout         = 300
   version         = var.external_dns_helm_chart_version
-  
+
```
Comment (EronWright) [inline external_dns.tf:99]: The external_dns module should depend on the istio module, since it relies on the CRDs.

File: variables.tf [hunk 3]
```diff
@@ -380,14 +374,8 @@ variable "istio_mesh_id" {
...
 
-variable "istio_namespace" {
-  default     = "sn-system"
-  description = "The namespace used for installing the Istio components."
-  type        = string
-}
-
 variable "istio_network" {
-  default     = null
+  default     = "network1"
```
Comment (EronWright) [inline variables.tf:378]: I suppose the name `default` would be a reasonable alternative to `network1`.

**Review by EronWright — APPROVED** [review]

File: locals.tf (hunk 1 above)
Comment (jrsdav) [reply locals.tf:38]: Not a problem, I'll do this for `external_dns.tf` now and refactor the other locals into their respective resource files in a future PR.

File: external_dns.tf (hunk 2 above)
Comment (jrsdav) [reply external_dns.tf:99]: We can't add a dependency to Istio if it's an optional plugin (i.e. Platform use case).

Comment (EronWright) [reply external_dns.tf:105]: I suppose that the dependency would need to be expressed in the aggregated module, that I believe should be prescriptive.

**Pull request merged**